src/User/UserContainer.jsx

```diff
@@ -10,35 +10,72 @@ const UserContainer = () => {
   const [user, setUser] = useState(null);
   const [isLoading, setIsLoading] = useState(true);
   const [error, setError] = useState(false);
+  const [isCurrentUser, setIsCurrentUser] = useState(false);
+  const [managerName, setManagerName] = useState("");
   const { id } = useParams();
   const [cookies] = useCookies([cookieName]);
 
+  const checkIfCurrentUser = (fetchedId, currentUserId) => {
+    if (Number(fetchedId) === currentUserId) {
+      setIsCurrentUser(true);
+    }
+  };
+
+  const getManagerName = async managerId => {
+    if (managerId) {
+      try {
+        const manager = await getUserById(
+          managerId,
+          cookies.sessionToken.token
+        );
+        setManagerName(`${manager.firstName} ${manager.lastName}`);
+      } catch (error) {
+        setError(true);
+      }
+    } else {
+      setManagerName("None Assigned");
+    }
+  };
+
   const fetchData = async userId => {
     setIsLoading(true);
 
     try {
       const user = await getUserById(userId, cookies.sessionToken.token);
       setUser(user);
+      checkIfCurrentUser(userId, cookies.sessionToken.userId);
     } catch (error) {
       setError(true);
     }
+
     setIsLoading(false);
   };
 
   useEffect(() => {
     if (id) {
       fetchData(id);
     }
+    if (isCurrentUser) {
+      getManagerName(user.managerId);
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [id]);
+  }, [id, isCurrentUser]);
 
   return (
     <Switch>
-      <Route path="/users/edit/:id">
+      <Route path="/users/:id/edit">
         <UpdateProfile />
       </Route>
       <Route path="/users/:id">
-        <ProfilePage user={user} isLoading={isLoading} error={error} />
+        <ProfilePage
+          user={user}
+          isLoading={isLoading}
+          error={error}
+          isCurrentUser={isCurrentUser}
```

Comment:
this should be checked in the component, `user.id === path.id`?

Reply:
Now that we have the redirect should this still be checked in the component or in the container?

Reply:
In the component you are using it to check whether to render the edit link button, therefore some kind of check is still needed so the user doesn't click on an edit button when its not their profile

Reply:
Ah yeah don't worry we still have that, I was more asking abut if it should be done in situ or not. I've made a function to return the true or false which I call in the conditional rendering checks.